File: ckgame-java/src/main/java/com/ckgame/politics/CasusBelli.java
package com.ckgame.politics;

/**
 * 战争借口（最小化实现，对应 Python diplomacy.CasusBelli）。
 */
public enum CasusBelli {
    CLAIM,
    CONQUEST,
    INDEPENDENCE,
    DEPOSE_LIEGE,
    HOLY_WAR,
    DE_JURE,
    RIVALRY,
    SUBJUGATION,
    TRADE_WAR;

    /** 战争目标分数。 */
    public int warscoreGoal() {
        return this == CONQUEST ? 80 : 100;
    }

    /** 中文名称（供日志使用）。 */
    public String nameZh() {
        return switch (this) {
            case CLAIM -> "宣称战争";
            case CONQUEST -> "征服";
            case INDEPENDENCE -> "独立战争";
            case DEPOSE_LIEGE -> "废黜领主";
            case HOLY_WAR -> "圣战";
            case DE_JURE -> "法理战争";
            case RIVALRY -> "世仇战争";
            case SUBJUGATION -> "臣服战争";
            case TRADE_WAR -> "贸易战争";
        };
    }

    /** 宣战时消耗的威望。 */
    public double prestigeCost() {
        return switch (this) {
            case CLAIM -> 50.0;
            case CONQUEST -> 100.0;
            case INDEPENDENCE -> 0.0;
            case DEPOSE_LIEGE -> 150.0;
            case HOLY_WAR -> 0.0;
            case DE_JURE -> 75.0;
            case RIVALRY -> 25.0;
            case SUBJUGATION -> 200.0;
            case TRADE_WAR -> 50.0;
        };
    }

    /** 胜利时攻击者获得的威望。 */
    public double attackerPrestigeOnWin() {
        return switch (this) {
            case CLAIM -> 50.0;
            case CONQUEST -> 80.0;
            case INDEPENDENCE -> 100.0;
            case DEPOSE_LIEGE -> 120.0;
            case HOLY_WAR -> 150.0;
            case DE_JURE -> 60.0;
            case RIVALRY -> 40.0;
            case SUBJUGATION -> 200.0;
            case TRADE_WAR -> 60.0;
        };
    }
}

File: ckgame-ts/src/politics/CasusBelli.ts
/**
 * Casus belli (war justification) types.
 */
export enum CasusBelli {
  CLAIM = 'CLAIM',
  CONQUEST = 'CONQUEST',
  INDEPENDENCE = 'INDEPENDENCE',
  DEPOSE_LIEGE = 'DEPOSE_LIEGE',
  HOLY_WAR = 'HOLY_WAR',
  DE_JURE = 'DE_JURE',
  RIVALRY = 'RIVALRY',
  SUBJUGATION = 'SUBJUGATION',
  TRADE_WAR = 'TRADE_WAR',
}

const _nameZh: Record<CasusBelli, string> = {
  [CasusBelli.CLAIM]: '宣称战争',
  [CasusBelli.CONQUEST]: '征服',
  [CasusBelli.INDEPENDENCE]: '独立战争',
  [CasusBelli.DEPOSE_LIEGE]: '废黜领主',
  [CasusBelli.HOLY_WAR]: '圣战',
  [CasusBelli.DE_JURE]: '法理战争',
  [CasusBelli.RIVALRY]: '世仇战争',
  [CasusBelli.SUBJUGATION]: '臣服战争',
  [CasusBelli.TRADE_WAR]: '贸易战争',
};

const _warscoreGoal: Record<CasusBelli, number> = {
  [CasusBelli.CLAIM]: 80,
  [CasusBelli.CONQUEST]: 100,
  [CasusBelli.INDEPENDENCE]: 80,
  [CasusBelli.DEPOSE_LIEGE]: 100,
  [CasusBelli.HOLY_WAR]: 100,
  [CasusBelli.DE_JURE]: 80,
  [CasusBelli.RIVALRY]: 60,
  [CasusBelli.SUBJUGATION]: 100,
  [CasusBelli.TRADE_WAR]: 60,
};

const _prestigeCost: Record<CasusBelli, number> = {
  [CasusBelli.CLAIM]: 50,
  [CasusBelli.CONQUEST]: 100,
  [CasusBelli.INDEPENDENCE]: 0,
  [CasusBelli.DEPOSE_LIEGE]: 150,
  [CasusBelli.HOLY_WAR]: 0,
  [CasusBelli.DE_JURE]: 75,
  [CasusBelli.RIVALRY]: 25,
  [CasusBelli.SUBJUGATION]: 200,
  [CasusBelli.TRADE_WAR]: 50,
};

const _attackerPrestigeOnWin: Record<CasusBelli, number> = {
  [CasusBelli.CLAIM]: 50,
  [CasusBelli.CONQUEST]: 80,
  [CasusBelli.INDEPENDENCE]: 100,
  [CasusBelli.DEPOSE_LIEGE]: 120,
  [CasusBelli.HOLY_WAR]: 150,
  [CasusBelli.DE_JURE]: 60,
  [CasusBelli.RIVALRY]: 40,
  [CasusBelli.SUBJUGATION]: 200,
  [CasusBelli.TRADE_WAR]: 60,
};

export function cbNameZh(cb: CasusBelli): string {
  return _nameZh[cb] ?? cb;
}

export function cbWarscoreGoal(cb: CasusBelli): number {
  return _warscoreGoal[cb] ?? 100;
}

export function cbPrestigeCost(cb: CasusBelli): number {
  return _prestigeCost[cb] ?? 0;
}

export function cbAttackerPrestigeOnWin(cb: CasusBelli): number {
  return _attackerPrestigeOnWin[cb] ?? 0;
}
